tests/test_tags.py:43: AssertionError on Darwin

Links:


- https://hydra.nixos.org/build/298947878 (python312Packages.meson-python.aarch64-darwin)
- https://hydra.nixos.org/build/298947879 (python312Packages.meson-python.x86_64-darwin)
- https://hydra.nixos.org/build/298967891 (python313Packages.meson-python.aarch64-darwin)
- https://hydra.nixos.org/build/298967893 (python313Packages.meson-python.x86_64-darwin)


```python
________________________________ test_wheel_tag ________________________________

    def test_wheel_tag():
>       assert str(mesonpy._tags.Tag()) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
E       AssertionError: assert 'cp313-cp313-...sx_11_0_arm64' == 'cp313-cp313-...sx_15_0_arm64'
E         
E         - cp313-cp313-macosx_15_0_arm64
E         ?                     ^
E         + cp313-cp313-macosx_11_0_arm64
E         ?                     ^

tests/test_tags.py:43: AssertionError
____________________________ test_tag_platlib_wheel ____________________________

    def test_tag_platlib_wheel():
        builder = wheel_builder_test_factory({
            'platlib': [f'extension{SUFFIX}'],
        })
>       assert str(builder.tag) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
E       AssertionError: assert 'cp313-cp313-...sx_11_0_arm64' == 'cp313-cp313-...sx_15_0_arm64'
E         
E         - cp313-cp313-macosx_15_0_arm64
E         ?                     ^
E         + cp313-cp313-macosx_11_0_arm64
E         ?                     ^

tests/test_tags.py:99: AssertionError
_____________________________ test_tag_stable_abi ______________________________

    def test_tag_stable_abi():
        builder = wheel_builder_test_factory({
            'platlib': [f'extension{ABI3SUFFIX}'],
        }, limited_api=True)
        # PyPy does not support the stable ABI.
        abi = 'abi3' if '__pypy__' not in sys.builtin_module_names else ABI
>       assert str(builder.tag) == f'{INTERPRETER}-{abi}-{PLATFORM}'
E       AssertionError: assert 'cp313-abi3-macosx_11_0_arm64' == 'cp313-abi3-macosx_15_0_arm64'
E         
E         - cp313-abi3-macosx_15_0_arm64
E         ?                    ^
E         + cp313-abi3-macosx_11_0_arm64
E         ?                    ^

tests/test_tags.py:108: AssertionError
```

--- a/tests/test_tags.py
+++ b/tests/test_tags.py
@@ -40,7 +40,13 @@ SUFFIX = sysconfig.get_config_var('EXT_SUFFIX')
 ABI3SUFFIX = get_abi3_suffix()
 
 
-def test_wheel_tag():
+@pytest.fixture()
+def cleanenv(monkeypatch):
+    # $MACOSX_DEPLOYMENT_TARGET affects the computation of the platform tag on macOS.
+    monkeypatch.delenv('MACOSX_DEPLOYMENT_TARGET', raising=False)
+
+
+def test_wheel_tag(cleanenv):
     assert str(mesonpy._tags.Tag()) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
     assert str(mesonpy._tags.Tag(abi='abi3')) == f'{INTERPRETER}-abi3-{PLATFORM}'
 
@@ -112,14 +118,14 @@ def test_tag_purelib_wheel():
     assert str(builder.tag) == 'py3-none-any'
 
 
-def test_tag_platlib_wheel():
+def test_tag_platlib_wheel(cleanenv):
     builder = wheel_builder_test_factory({
         'platlib': [f'extension{SUFFIX}'],
     })
     assert str(builder.tag) == f'{INTERPRETER}-{ABI}-{PLATFORM}'
 
 
-def test_tag_stable_abi():
+def test_tag_stable_abi(cleanenv):
     builder = wheel_builder_test_factory({
         'platlib': [f'extension{ABI3SUFFIX}'],
     }, limited_api=True)
@@ -130,7 +136,7 @@ def test_tag_stable_abi():
 
 @pytest.mark.xfail(sys.version_info < (3, 8) and sys.platform == 'win32', reason='Extension modules suffix without ABI tags')
 @pytest.mark.xfail('__pypy__' in sys.builtin_module_names, reason='PyPy does not support the stable ABI')
-def test_tag_mixed_abi():
+def test_tag_mixed_abi(cleanenv):
     builder = wheel_builder_test_factory({
         'platlib': [f'extension{ABI3SUFFIX}', f'another{SUFFIX}'],
     }, pure=False, limited_api=True)

